mysql_tests/test_iterate.py

```diff
@@ -0,0 +1,90 @@
+from gino import UninitializedError
+import pytest
+
+from .models import db, User
+
+pytestmark = pytest.mark.asyncio
+
+
+@pytest.fixture
+def names(sa_engine):
+    rv = {"11", "22", "33"}
+    sa_engine.execute(User.__table__.insert(), [dict(name=name) for name in rv])
+    yield rv
+    sa_engine.execute("DELETE FROM gino_users")
+
+
+# noinspection PyUnusedLocal,PyShadowingNames
+async def test_bind(bind, names):
+    with pytest.raises(ValueError, match="No Connection in context"):
+        async for u in User.query.gino.iterate():
+            assert False, "Should not reach here"
+    with pytest.raises(ValueError, match="No Connection in context"):
+        await User.query.gino.iterate()
+    with pytest.raises(ValueError, match="No Connection in context"):
+        await db.iterate(User.query)
+
+    result = set()
+    async with bind.transaction():
+        async for u in User.query.gino.iterate():
+            result.add(u.nickname)
+    assert names == result
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/python-gino/gino/pullRequest?prid=5587950)